### PR TITLE
Created architecture for the store

### PR DESCRIPTION
**Pull request by JorgeGarciaIrazabal**
Added the store and come basic architecture.

Using `mobx` we can pass the `store` as a property in every page and change it from there. Any modification will render again the view (just the modified elements)

The only thing we have to remember is to **add `@observable` in any attribute of the store** and **`@observer` to every `container` or `page`** that will change the store.**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,8 +9,10 @@
   "dependencies": {
```
Comment (JorgeGarciaIrazabal) [inline package.json:9]: adding some dependencies we will use in the future, not a bid deal

File: src/app.js [hunk 2]
```diff
@@ -4,18 +4,20 @@ import {COLOR, ThemeProvider, Toolbar} from 'react-native-material-ui';
 import routes from './routes';
 import core_styles from './styles/core-styles';
 import theme from './styles/ui-theme';
+import store from './store/store';
 
 const UIManager = NativeModules.UIManager;
 
 export default class App extends Component {
-  static configureScene(route) {
```
Comment (JorgeGarciaIrazabal) [inline src/app.js:13]: Removing the static because I prefer to have normal functions. No need to have it as static

File: src/models/contact.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import {observable} from 'mobx';
```
Comment (JorgeGarciaIrazabal) [inline src/models/contact.js:1]: First stage of the models we are going to use. I prefer to have them as classes better than normal javascript objects `{}`.

This is just my personal option, any thoughts?

File: src/models/contact.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import React from 'react';
+
+export default class Contact {
+  /** @type string*/
+  @observable name;
```
Comment (JorgeGarciaIrazabal) [inline src/models/contact.js:6]: remember to add `@observable` if it is going to be in the store

File: src/pages/login.js [hunk 5]
```diff
@@ -6,10 +6,9 @@ import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome';
 import {Sae} from 'react-native-textinput-effects';
 import {Button} from 'react-native-material-ui';
 import theme from '../styles/ui-theme';
+import {observer} from 'mobx-react/native';
 
-
-
-
+@observer
```
Comment (JorgeGarciaIrazabal) [inline src/pages/login.js]: remember to add `@observer` if it is going to change the store

File: src/app.js [hunk 6]
```diff
@@ -25,11 +27,13 @@ export default class App extends Component {
           <route.Page
             route={route}
             navigator={navigator}
+            store={store}
```
Comment (JorgeGarciaIrazabal) [inline src/app.js:32]: we pass here the `store` as a `prop`. To get it from the login page for instance, we just need to do 
```
let {self} = this.props.store;
```
if we want to get self from the `store`

**Comment by JorgeGarciaIrazabal** [comment]
This PR might have some conflicts with yours so I will do a rebase and fix the conflicts whenever you merge yours

**Comment by Fdom92** [comment]
Awesome! Ill merge the other one asap :)

**Pull request merged**